crates/experimentation_client/src/lib.rs

```diff
@@ -81,6 +85,28 @@ impl Client {
                         );
                     }
                 };
+
+                let experiment_groups = get_experiment_groups(
+                    hostname.clone(),
+                    self.http_client.clone(),
+                    *start_date,
+                    self.client_config.tenant.to_string(),
+                )
+                .await;
+                match experiment_groups {
+                    Ok(experiment_groups) => {
+                        let mut exp_group_store = self.experiment_groups.write().await;
+                        *exp_group_store = experiment_groups;
+                    }
+                    Err(e) => {
+                        log::error!(
+                            "Failed to fetch experiments from the server with error: {}",
+                            e
+                        );
+                    }
+                };
+
+                *start_date = Utc::now();
```

Comment:
even on error , this start_date will get updates , that is not a good experience